### DISCUSSION TITLE
Enabling jinx using a directory variable

### DISCUSSION THREAD

**Discussion by arapede** [minad/jinx#158] [Q&A] [answered]
Hi! I raised a question on reddit the other day about enabling `jinx` using a file-local variable. Thanks for the speedy fix! I think there's a similar issue enabling it with directory variables. 

I create an empty directory, create a file `.dir-locals.el` with the contents `((nil . ((mode . jinx))))`, and then create a new file `test.txt`. When I open the last of these I get an error: `Directory-local variables error: (error Lisp nesting exceeds ‘max-lisp-eval-depth’)`.

Thanks for all the great work.

(`GNU Emacs 29.1 (build 2, x86_64-pc-linux-gnu, GTK+ Version 3.24.38, cairo version 1.18.0) of 2023-11-05`, `jinx 20240311.821`)

**Comment by minad** [comment]
Thanks for the report. Please check if https://github.com/minad/jinx/commit/1e25c7dac958b9698ab854322c9bbdf09c919c93 fixes the issue. Otherwise I think I may have to use a different approach for the local variables, since the API is just too brittle.

**Comment by arapede** [comment]
Yes, I just tried it and it seems to work just fine. Many thanks again for resolving both issues so fast!

**Reply by minad** [reply, marked as answer]
Unfortunately I had to revert the fix again, due to the more important issue https://github.com/minad/jinx/issues/161.

EDIT: Pushed another fix for this problem, see https://github.com/minad/jinx/commit/d4c67d28578e0493f90786a1bf0c5fb4cf12ba82.

**Reply by arapede** [reply]
Brilliant. Just to confirm that the latest fix still works for me.
